src/media/media.ts

```diff
@@ -0,0 +1,820 @@
+import { EventEmitter } from 'events';
+import StrictEventEmitter from 'strict-event-emitter-types';
+import { cloneDeep } from 'lodash';
+import browserama from 'browserama';
+
+import GenesysCloudWebrtcSdk from '../client';
+import { IExtendedMediaSession, IMediaRequestOptions, MediaState, SdkMediaEvents } from '../types/interfaces';
+import { createAndEmitSdkError } from '../utils';
+import { SdkErrorTypes } from '../types/enums';
+
+declare var window: {
+  navigator: {
+    mediaDevices: {
+      getDisplayMedia?: (constraints: MediaStreamConstraints) => Promise<MediaStream>;
+    } & MediaDevices;
+  } & Navigator;
+  webkitAudioContext: typeof AudioContext;
+} & Window & typeof globalThis;
+
+export class SdkMedia extends (EventEmitter as { new(): StrictEventEmitter<EventEmitter, SdkMediaEvents> }) {
+  private sdk: GenesysCloudWebrtcSdk;
+  private state: MediaState;
+  private audioTracksBeingMonitored: { [key: string]: any } = {};
+  private allMediaTracksCreated: MediaStreamTrack[] = [];
```

Comment:
We will probably need to clean this up somehow. I know the purescale peeps will complain about this memory leak. Realistically it's not a huge leak for end users. Is there a reason we keep track of *all* tracks rather than just the active ones?